### PR TITLE
Feature/project permissions

### PR DESCRIPTION
**Pull request by dcl10**
# Changes
- Added `ProjectListView` which shows all projects. No additional permissions required.
  - found at `/projects`
- Added `ProjectRetrieveView` which shows a specific project but only if the user is a member of that project.
  - found at `/projects/<int:pk>`
- Added permission class `CanViewProject` which checks if a user is a member of the project they are trying to view.

# Screenshots
### `/projects`
Screenshot 2022-02-02 at 09 33 06

### `/projects/<non-member project>`
Screenshot 2022-02-02 at 09 33 56

### `/projects/<member project>`
Screenshot 2022-02-02 at 09 36 19

### CODE REVIEW

**Review by spco — COMMENTED** [review]
In general looks good. Can you perhaps show this working in action before we merge to devel please? There are enough moving parts on this and all the following PRs for permissions that I'm sure we are going to miss things. Going through with demos is most likely to catch issues early. I'm rubbish at understanding whether you've done everything necessary in this PR.

File: api/mapping/views.py [hunk 1]
```diff
@@ -988,11 +1015,7 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
         x = ScanReport.objects.get(pk=self.kwargs.get("pk"))
-        context.update(
-            {
-                "scan_report": x,
-            }
-        )
+        context.update({"scan_report": x})
```
Comment (calmacx) [inline api/mapping/views.py:1018]: odd that this doesnt fail the black job (?)

**Review by calmacx — APPROVED** [review]

**Comment by calmacx** [comment]
LGTM... just will wait until tomorrow morning to demo as @spco  suggested before merge to devel branch ?

File: api/mapping/views.py (hunk 1 above)
Comment (dcl10) [reply api/mapping/views.py:1019]: The trailing comma at the end of line 993 was what caused it to spread out

**Comment by spco** [comment]
All good by me now - will need to run the unorphan_scanreports command on `ccnet-dev-db` to correctly associate `Datasets` to `Projects`.

**Pull request merged**